File: 준영/실패율.py
def solution(N, stages):
    answer = []
    arr = [[0, 0, i]for i in range(N)]

    for i in stages:
        for j in range(0, i-1):
            arr[j][0] += 1
            
    div = len(stages)

    for i in range(N):
        if div == 0:
            arr[i][1] = 0
        else:
            arr[i][1] = (div - arr[i][0]) / div
        div = arr[i][0]

    arr.sort(key=lambda x: (-x[1]))

    for i in range(N):
        answer.append(arr[i][2]+1)
    return answer

print(solution(5, [2, 1, 2, 6, 2, 4, 3, 3]))

File: 준영/실패율.js
function solution(N, stages) {
  var answer = []
  var arr = Array.from({length: N}, x => 0)
  for (const i of stages) {
    for (let j = 0; j < i-1; j++) {
      arr[j] += 1
    }
  }

  var fail = Array.from({length: N}, (x, i) => [0, i+1])

  var div = stages.length

  for (let i = 0; i < N; i++) {
    if (div === 0) {
      fail[i][0] = 0
    } else {
      fail[i][0] = (div - arr[i]) / div
    }
    div = arr[i]
  }
  
  fail.sort(function (a, b) {
    return b[0] - a[0]
  })

  answer = fail.map(x => x[1])
  return answer
}

console.log(solution(5, [2, 1, 2, 6, 2, 4, 3, 3]))